client/verta/verta/deployment/update/rules.py

```diff
@@ -24,6 +25,26 @@ def _as_json(self):
             ],
         }
 
+    @staticmethod
+    def _from_json(json_str):
+        rule_dict = json.loads(json_str)
+        rule_id = int(rule_dict["rule_id"])
+        rule_name = rule_dict['rule_parameters'][0]['name']
+        rule_value = float(rule_dict['rule_parameters'][0]['value'])
+
+        if rule_name == "latency_avg":
+            rule = AverageLatencyThresholdRule(rule_value)
+        elif rule_name == "latency_p90":
+            rule = P90LatencyThresholdRule(rule_value)
+        elif rule_name == "error_rate":
+            rule = ErrorRateRule(rule_value)
+        else:
+            raise ValueError("no rule with name {} exists".format(rule_name))
```

Comment:
Could we re-write this to use a mapping that checks the classes' `_NAME` directly?
Something like
```python
for Subclass in (...):
    if rule_name == Subclass._NAME:
        rule = Subclass(rule_value)
        break
else:
    raise ValueError("no rule with name {} exists".format(rule_name))
```
That way, if we ever change the name of a rule, we'd only have to change it in one place (the class attr).

Reply:
Updated! See if the new _from_json makes sense.
I separate the core logic into _from_dict because I want to reuse it for update_from_config.